### PR TITLE
Add Autopublish Support

### PR DESCRIPTION
**Pull request by Croebh**
`.autopublish`
`.autopublish_add <channel mention>`
`.autopublish_remove <channel mention>`

Simply allows you to add channels to auto-publish posts to reduce the need for manual publishing. The bot will need Manage Messages and Send Messages permissions in the added channels in order to publish.

### CODE REVIEW

**Comment by Croebh** [comment]
@zhudotexe requesting review!

https://stackoverflow.com/questions/32262295/adding-a-reviewer-to-a-github-pull-request/41028889?noredirect=1#comment91586105_41028889 supposedly pinging you is the only option D:

File: cogs/autopublish.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+class AutoPublish(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self.autopublish_channels = bot.db.jget("autopublish", {})
+
+    @commands.Cog.listener()
+    async def on_message(self, message):
+        if not self.bot.is_ready():
+            return
+        if not message.channel.id in self.autopublish_channels:
```
Comment (zhudotexe) [inline cogs/autopublish.py]: nit: python code style for this is `not in`

File: cogs/autopublish.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    # ==== commands ====
+    @commands.group(invoke_without_command=True)
+    @commands.has_role(constants.MOD_ROLE_ID)
+    async def autopublish(self, ctx):
+        """Shows all channels with active autopublish."""
+        embed = discord.Embed(colour=0x60AFFF, title="Active Autopublish Channels")
+        embed.description = '\n'.join(f"<#{channel}>"
+                                      for channel in self.autopublish_channels) \
+                            or "No active channels."
+        embed.set_footer(text="Use \".autopublish add #channel #\" to add a channel rule, "
```
Comment (zhudotexe) [inline cogs/autopublish.py]: looks like you've got an extraneous `#` at the end here

**Review by zhudotexe — APPROVED** [review]
LGTM with minor nits

File: cogs/autopublish.py (hunk 2 above)
Comment (Croebh) [reply cogs/autopublish.py]: Ah sure did, from when I modified this from .autodelete which had days

**Pull request merged**